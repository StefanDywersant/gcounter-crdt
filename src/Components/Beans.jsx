import React, { Children } from 'react';
import PropTypes from 'prop-types';
import styles from './Beans.scss';


/**
 * Renders beans list
 */
const Beans = ({children, className}) => (
	<ul className={[styles.beans, className].join(' ')}>
		{Children.map(
			children,
			child => <li>{child}</li>
		)}
	</ul>
);


Beans.propTypes = {

	/**
	 * Optional class name for outer element
	 */
	className: PropTypes.string

};


export default Beans;