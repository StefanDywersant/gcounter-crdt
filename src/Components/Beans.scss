.beans {
	padding: 0;

	display: flex;
	flex-wrap: wrap;
	align-content: flex-start;

	> li {
		list-style: none;
	}
}