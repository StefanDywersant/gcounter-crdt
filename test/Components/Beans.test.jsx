import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Beans from '../../src/Components/Beans';

const renderer = new ReactShallowRenderer();

describe('Component rendering', () => {

	it('renders correctly with default props', () => {
		expect(renderer.render(
			<Beans />
		)).toMatchSnapshot();
	});

	it('renders correctly with className set', () => {
		expect(renderer.render(
			<Beans className="customClassName" />
		)).toMatchSnapshot();
	});

	it('renders correctly with children', () => {
		expect(renderer.render(
			<Beans>
				<div />
				<p />
			</Beans>
		)).toMatchSnapshot();
	});

});
