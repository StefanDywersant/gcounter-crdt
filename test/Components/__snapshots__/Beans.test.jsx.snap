// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendering renders correctly with children 1`] = `
<ul
  className="beans "
>
  <li>
    <div />
  </li>
  <li>
    <p />
  </li>
</ul>
`;

exports[`Component rendering renders correctly with className set 1`] = `
<ul
  className="beans customClassName"
/>
`;

exports[`Component rendering renders correctly with default props 1`] = `
<ul
  className="beans "
/>
`;
